### PR TITLE
adding a simple function to get prime factors for a number

### PR DESCRIPTION
**Pull request by aelnaiem**
[asana task](https://app.asana.com/0/201972293666219/205201279084528)

### CODE REVIEW

File: test/Negative.test.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ function testNegate(exprString, outputStr) {
   });
 }
 
-describe('negatePolynomialTerm', function() {
+describe('negate', function() {
```
Comment (evykassirer) [inline test/Negative.test.js]: thanks!

File: test/solveEquation.test.js [hunk 2]
```diff
@@ -69,15 +69,15 @@ describe('solveEquation for non = comparators', function() {
...
   const leftNode = math.parse(sides[0]);
   const rightNode = math.parse(sides[1]);
 
   const steps = solveEquation(leftNode, rightNode, comparator, debug);
-  const acualExplanation = steps[steps.length -1].explanation;
-  it(equationString + ' -> ' + expectedExplanation, function () {
-    assert.equal(acualExplanation, expectedExplanation);
+  const acualChange = steps[steps.length -1].changeType;
+  it(equationString + ' -> ' + expectedChange, function () {
+    assert.equal(acualChange, expectedChange);
```
Comment (evykassirer) [inline test/solveEquation.test.js]: *actual

File: lib/Factor.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  const root = Math.sqrt(number);
+  let candidate = 2;
+  if (number % 2) {
+    candidate = 3; // assign first odd
+    while (number % candidate && candidate <= root) {
+      candidate = candidate + 2;
+    }
+  }
+
+  // if no factor found then number is prime
```
Comment (evykassirer) [inline lib/Factor.js]: or 1 or 0 ;D

Comment (evykassirer) [inline lib/Factor.js]: ^ what a nerd

**Comment by evykassirer** [comment]
mostly comments about your comments, but otherwise looks good!

File: lib/Factor.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  if (candidate > root) {
+    factors.push(number);
+  }
+  else {
+    // if number isn't prime factor make recursive call
```
Comment (aelnaiem) [inline lib/Factor.js]: No if it's not, then we want to know what the prime factors of it are.

File: lib/Factor.js [hunk 5]
```diff
@@ -34,4 +34,18 @@ Factor.getPrimeFactors = function(number){
   return factors;
 };
 
+Factor.getFactorPairs = function(number){
```
Comment (evykassirer) [inline lib/Factor.js]: oh yeah! we'll need this for quadratics

docstring here? also should we handle number being negative here or before calling this?

File: lib/Factor.js (hunk 4 above)
Comment (evykassirer) [inline lib/Factor.js]: ohhh, got it

can you make the comment "if the factor isn't prime, make a recursive call to add its prime factors" or something like that? I thought number was `number`

**Comment by aelnaiem** [comment]
I also added a function to get factor pairs (which will eventually be used in factoring binomials)

File: lib/Factor.js (hunk 4 above)
Comment (aelnaiem) [inline lib/Factor.js]: Number is `number`. Basically, if we find a prime factor, we divide the number by that prime factor and get more prime factors. I'll add that though

Comment (evykassirer) [inline lib/Factor.js]: oh, yeah that makes sense

**Pull request merged**